wrphttp/decoders_test.go

```diff
@@ -196,3 +197,132 @@ func TestDecodeRequestHeaders(t *testing.T) {
 	t.Run("Success", testDecodeRequestHeadersSuccess)
 	t.Run("Invalid", testDecodeRequestHeadersInvalid)
 }
+
+func testDecodeRequestSuccess(t *testing.T) {
```

Comment:
Not a blocker, but we should start moving consistently to github.com/stretchr/testify/suite for test suites.  That makes these kinds of subtests much clearer and easier to implement and maintain.